File: java/program34.java
import java.util.*;

class Number 
{
    public int iBase;
    public int iPower;

    Number(int no1,int no2)
    {
        iBase = no1;
        iPower = no2;
    }

    public int CalculatePower()
    {
        int iResult =  1;

        for(int iCnt = 1; iCnt <= iPower; iCnt++)
        {
            iResult = iResult * iBase;
        }
        return iResult;
    }
}

class program34
{
    public static void main(String arg[])
    {
        int iValue1 = 0, iValue2 = 0,iRet = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter Base ");
        iValue1 = sobj.nextInt();

        System.out.println("Enter Power ");
        iValue2 = sobj.nextInt();

        Number obj = new Number(iValue1,iValue2);
        iRet = obj.CalculatePower();

        System.out.println("Result is : "+iRet);
    }
}

File: cpp/program34.cpp
// program32.c

#include<iostream>
using namespace std;

typedef unsigned long int ULINT;

class Number
{
    public:
        int iBase;
        int iPower;

    Number(int X, int Y)
    {
        iBase = X;
        iPower = Y;
    }

    ULINT CalculatePower()
    {
        int iCnt = 0;
        ULINT iResult = 1;

        for(iCnt = 1; iCnt <= iPower; iCnt++ )
        {
            iResult = iResult * iBase;
        }
        return iResult;
    }
    
};

int main()
{
    int iValue1 = 0;
    int iValue2 = 0;
    ULINT iRet = 0;

    cout<<"Enter Base"<<"\n";
    cin>>iValue1;

    cout<<"Enter power"<<"\n";
    cin>>iValue2;

    Number nobj(iValue1,iValue2);

    iRet = nobj.CalculatePower();
    cout<<"Result is : "<<iRet<<"\n";
    
    return 0;
}

